metricbeat/helper/kubernetes/state_metricset.go

```diff
@@ -0,0 +1,113 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kubernetes
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	k8smod "github.com/elastic/beats/v7/metricbeat/module/kubernetes"
+	"github.com/elastic/beats/v7/metricbeat/module/kubernetes/util"
+)
+
+const prefix = "state_"
+
+var mappings = map[string]*prometheus.MetricsMapping{}
```

Comment:
I see this map is still not documented. Could you add a docsting here to explain what kind of information this variable holds and an example. 

Also, are we sure if this map would be able to handle concurrent read/writes properly? In general, I don't think that "global" variables is a good pattern to follow. However if we are sure that we want to go down this specific path of abstraction here I don't see how we could avoid this global variable. @gizas any thoughts?

Reply:
For now I added a lock to prevent concurrent map read/write. I followed the same approach [as the registry](https://github.com/elastic/beats/blob/21b6128c9578f066d7503c2b2419b9be3db35296/metricbeat/mb/registry.go#L199-L200).

Reply:
I tried to study a way to pass the mapping as a variable to the New function, but unfortunately, I don't think we can do that without changing some things outside of our module.